compiler/src/dotty/tools/dotc/core/TypeOps.scala

```diff
@@ -560,6 +560,9 @@ object TypeOps:
             widenUnions = tp.widenUnions)(using mapCtx)
           val lo1 = apply(lo)
           if (lo1 ne lo) lo1 else tp
+        case tp @ ConstantType(Constant(tp1: Type)) =>
+          val tp2 = apply(tp1)
+          if tp2 eq tp1 then tp else ConstantType(Constant(tp2))
```

Comment:
We can't soundly approximate the argument of `classOf`, it would give us another class (or even something that isn't a valid classOf argument such as an intersection). I think the best we can do is return an empty Range and emit an error (also this could probably be done in ApproximatingTypeMap directly).

Reply:
Actually I guess we can approximate the type of `classOf` to be "some unknown subtype of `Class[? <: approx(tp1)]`".
In any case, it seems that TypeMap is missing a case for `ConstantType` which is why these things have been ignored so far.